lib/stream.c

```diff
@@ -166,27 +163,33 @@ struct stream *stream_dupcat(struct stream *s1, struct stream *s2,
 	return new;
 }
 
-size_t stream_resize(struct stream *s, size_t newsize)
+size_t stream_resize_inplace(struct stream **sptr, size_t newsize)
 {
-	uint8_t *newdata;
-	STREAM_VERIFY_SANE(s);
+	struct stream *orig = *sptr;
 
-	newdata = XREALLOC(MTYPE_STREAM_DATA, s->data, newsize);
+	STREAM_VERIFY_SANE(orig);
 
-	if (newdata == NULL)
-		return s->size;
+	orig = XREALLOC(MTYPE_STREAM, orig, sizeof(struct stream) + newsize);
 
-	s->data = newdata;
-	s->size = newsize;
+	orig->size = newsize;
 
-	if (s->endp > s->size)
-		s->endp = s->size;
-	if (s->getp > s->endp)
-		s->getp = s->endp;
+	if (orig->endp > orig->size)
+		orig->endp = orig->size;
+	if (orig->getp > orig->endp)
+		orig->getp = orig->endp;
 
-	STREAM_VERIFY_SANE(s);
+	STREAM_VERIFY_SANE(orig);
 
-	return s->size;
+	*sptr = orig;
+	return orig->size;
+}
+
+size_t __attribute__((deprecated))stream_resize_orig(struct stream *s,
```

Comment:
I guess I shoulda been clearer - the attribute needs to be in the header file, otherwise gcc/clang won't be aware of it when they stumble over some random code that uses `stream_resize`